### PR TITLE
[Scripts] More tests for test_all

### PR DESCRIPTION
**Pull request by randallli**
so the release does not get hiccups like this past release.

### CODE REVIEW

**Comment by ianegordon** [comment]
test_all is currently used by Travis CI.
These tests are not compatible with Travis CI.
That said, our CI isn't as solid as it could be.

Are we okay with this?

**Comment by randallli** [comment]
oh didn't realize that it was run there. I want something for the release cop to run.

**Review by ianegordon — CHANGES_REQUESTED** [review]
I'm very pro-test but lets discuss in person before we merge.

**Comment by ianegordon** [comment]
super-nit: Consider avoiding two scripts with the same name. (Chances of me invoking the wrong one or getting the two confused are non-zero) 
test_release, check_release,  or verify_cocoapods perhaps?  

LGTM.

**Review by ianegordon — APPROVED** [review]

File: contributing/releasing.md [hunk 1]
```diff
@@ -60,7 +60,7 @@ test the release.
 
     scripts/prep_all
     scripts/build_all
-    scripts/test_all
```
Comment (ianegordon) [inline contributing/releasing.md]: We will still need scripts/test_all
Here and below.

File: scripts/release/test_all [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+readonly SCRIPTS_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+# run the normal test_all script
+("$SCRIPTS_DIR"/../test_all)
```
Comment (ianegordon) [inline scripts/release/test_all]: I would keep these scripts independent.

**Review by ianegordon — CHANGES_REQUESTED** [review]

**Review by ajsecord — CHANGES_REQUESTED** [review]
I don't think this change is the right one. `scripts/test_all` is supposed to do just that—run all the tests. It should be useable by developers, Travis CI, release cops, whoever. It's not release-specific.

File: scripts/release/test_all (hunk 2 above)
Comment (ajsecord) [inline scripts/release/test_all]: Name this script something descriptive, like `test_cocoapods_release`.

File: scripts/release/test_all [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+all_tests_ok=$?
+
+# extra tests for the release
+echo checking pod push authorization ...
```
Comment (ajsecord) [inline scripts/release/test_all]: Sentence case for everything.

File: scripts/release/test_all [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+echo lint podspec ...
+log_file="test_log_for_pod_lint.txt"
+if pod lib lint MaterialComponents.podspec >"$log_file" 2>&1; then
+    rm "$log_file"
+else
+  all_tests_ok=0
+    echo
+    echo "Failed to lint MaterialComponents.podspec"
+    guess_failure "$log_file"
```
Comment (ajsecord) [inline scripts/release/test_all]: Is `guess_failure` defined here? How was this tested?

File: scripts/release/test_all [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+echo checking pod push authorization ...
+if ! pod trunk me | grep -q 'MaterialComponents'; then
+  all_tests_ok=0
+  echo
+  echo "Failed! Register an account and session: https://guides.cocoapods.org/making/getting-setup-with-trunk.html"
+  echo "Continuing with next test..."
+  echo
+fi
+
+echo lint podspec ...
```
Comment (ajsecord) [inline scripts/release/test_all]: You might want to break this into two scripts: `test_pod_push` and `test_pod_format` or something. If I were making changes to the pod files I could see wanting to lint things even if I wasn't authorized to do a push.

**Review by ajsecord — CHANGES_REQUESTED** [review]

**Comment by ajsecord** [comment]
Let's drop this PR or push it through.

File: contributing/releasing.md (hunk 1 above)
Comment (randallli) [reply contributing/releasing.md]: Done.

File: scripts/release/test_all (hunk 3 above)
Comment (randallli) [reply scripts/release/test_all]: ack. Dropped echo.

File: scripts/release/test_all (hunk 2 above)
Comment (randallli) [reply scripts/release/test_all]: ack. Dropped umbrella test.

File: scripts/release/test_all (hunk 5 above)
Comment (randallli) [reply scripts/release/test_all]: Broke up into two as suggested. One ended up just being a subcommand so I put that back into the releaseing.md

File: scripts/release/test_all (hunk 4 above)
Comment (randallli) [reply scripts/release/test_all]: ack. Deleted these lines.

File: contributing/releasing.md [hunk 6]
```diff
@@ -192,18 +192,15 @@ Commit the results to your branch:
...
     pod lib lint MaterialComponents.podspec
 
-CocoaPods publishes a directory of publicly available pods through its **trunk** service.
-Note: Ensure that you can [push the podspec](#publish-to-cocoapods) later by checking for `MaterialComponents` in your list of available `Pods` when you:
+#### Check that you have push access to publish our cocoapod
 
-    pod trunk me
-
-If this fails or MaterialComponents is not listed [register an account and session](https://guides.cocoapods.org/making/getting-setup-with-trunk.html).
+    scripts/release/test_pod_push
```
Comment (ajsecord) [inline contributing/releasing.md]: Can we move this to some "setup"-type place? Every engineer doing releases should do this, but only once per machine, I believe. Having it here to be read and skipped over every time isn't great. If you can't find a better place, at the top in the "Before you start" section is fine.

File: scripts/test_all [hunk 7]
```diff
@@ -16,7 +16,7 @@
 
 readonly SCRIPTS_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 readonly ROOT_DIR="$(dirname $SCRIPTS_DIR)"
-# Note that the following is using 10.1 until Travis CI fixes 
+# Note that the following is using 10.1 until Travis CI fixes
 # https://github.com/travis-ci/travis-ci/issues/7031
 readonly DESTINATION="platform=iOS Simulator,name=iPhone 7,OS=10.1"
```
Comment (ajsecord) [inline scripts/test_all]: I think you need to sync, this line changed recently.

**Review by ajsecord — CHANGES_REQUESTED** [review]

File: contributing/releasing.md (hunk 6 above)
Comment (randallli) [reply contributing/releasing.md]: Done.

File: scripts/test_all (hunk 7 above)
Comment (randallli) [reply scripts/test_all]: Done.

**Comment by ajsecord** [comment]
This seems to be dead, dropping it.

**Pull request closed without merging**